src/Templating/TemplateRegistry.php

```diff
@@ -15,107 +15,22 @@
 
 /**
  * @author Timo Bakx <timobakx@gmail.com>
+ *
+ * NEXT_MAJOR: remove `MutableTemplateRegistryInterface` implementation with setter methods
  */
-final class TemplateRegistry implements MutableTemplateRegistryInterface
+final class TemplateRegistry extends BaseTemplateRegistry implements MutableTemplateRegistryInterface
 {
-    public const TYPE_ARRAY = 'array';
-    public const TYPE_BOOLEAN = 'boolean';
-    public const TYPE_DATE = 'date';
-    public const TYPE_TIME = 'time';
-    public const TYPE_DATETIME = 'datetime';
-    /**
-     * NEXT_MAJOR: Remove this constant.
-     *
-     * @deprecated since sonata-project/admin-bundle 3.68, to be removed in 4.0. Use Sonata\AdminBundle\Templating\TemplateRegistry::TYPE_STRING instead.
-     */
-    public const TYPE_TEXT = 'text';
-    public const TYPE_TEXTAREA = 'textarea';
-    public const TYPE_EMAIL = 'email';
-    public const TYPE_TRANS = 'trans';
-    public const TYPE_STRING = 'string';
-    /**
-     * NEXT_MAJOR: Remove this constant.
-     *
-     * @deprecated since sonata-project/admin-bundle 3.68, to be removed in 4.0. Use Sonata\AdminBundle\Templating\TemplateRegistry::TYPE_INTEGER instead.
-     */
-    public const TYPE_SMALLINT = 'smallint';
-    /**
-     * NEXT_MAJOR: Remove this constant.
-     *
-     * @deprecated since sonata-project/admin-bundle 3.68, to be removed in 4.0. Use Sonata\AdminBundle\Templating\TemplateRegistry::TYPE_INTEGER instead.
-     */
-    public const TYPE_BIGINT = 'bigint';
-    public const TYPE_INTEGER = 'integer';
     /**
-     * NEXT_MAJOR: Remove this constant.
-     *
-     * @deprecated since sonata-project/admin-bundle 3.68, to be removed in 4.0. Use Sonata\AdminBundle\Templating\TemplateRegistry::TYPE_FLOAT instead.
+     * NEXT_MAJOR: remove this method.
      */
-    public const TYPE_DECIMAL = 'decimal';
-    public const TYPE_FLOAT = 'float';
-    public const TYPE_IDENTIFIER = 'identifier';
-    public const TYPE_CURRENCY = 'currency';
-    public const TYPE_PERCENT = 'percent';
-    public const TYPE_CHOICE = 'choice';
-    public const TYPE_URL = 'url';
-    public const TYPE_HTML = 'html';
-    public const TYPE_MANY_TO_MANY = 'many_to_many';
-    public const TYPE_MANY_TO_ONE = 'many_to_one';
-    public const TYPE_ONE_TO_MANY = 'one_to_many';
-    public const TYPE_ONE_TO_ONE = 'one_to_one';
-
-    /**
-     * @var array<string, string>
-     */
-    private $templates = [];
-
-    /**
-     * @param string[] $templates
-     */
-    public function __construct(array $templates = [])
-    {
-        $this->templates = $templates;
-    }
-
-    public function getTemplates(): array
-    {
-        return $this->templates;
-    }
-
     public function setTemplates(array $templates)
```

Comment:
If this method need to be remove in next major, we should add `@deprecated` comment

And we should trigger a deprecation inside the method.

Reply:
Done.